soccer/src/soccer/strategy/agent/position/marker.cpp

```diff
@@ -0,0 +1,44 @@
+#include "marker.hpp"
+
+namespace strategy {
+Marker::Marker(u_int8_t robot_id) {}
+
+std::optional<RobotIntent> Marker::get_task(RobotIntent intent, const WorldState* world_state,
+                                            FieldDimensions field_dimensions) {
+    rj_geometry::Point targetPoint = world_state->get_robot(false, target).pose.position();
+    rj_geometry::Point ballPoint = world_state->ball.position;
+    rj_geometry::Point targetToBall = (ballPoint - targetPoint).normalized(0.55f);
+    planning::LinearMotionInstant goal{targetPoint + targetToBall, rj_geometry::Point{0.0, 0.0}};
+    intent.motion_command =
+        planning::MotionCommand{"path_target", goal, planning::FaceBall{}, true};
+    return intent;
+}
+
+int Marker::choose_target(WorldState* ws) {
+    // TODO: (James Vogt, github: jvogt23)
+    // If we ever use multiple Markers, they should choose different
+    // robots to track from each other. Logic for this operation must be
+    // added because multiple markers currently mark the same robot.
+    for (int i = 0; i < 11; i++) {
+        if (std::fabs(ws->get_robot(false, i).pose.position().x()) < 2.5 &&
```

Comment:
what is the `2.5` constant? are we choosing targets only on the edge of the field?

Reply:
This is actually to ignore targets that are right on the edge of the field. This is mostly so we don't accidentally mark a robot on the sideline or something like that.

Reply:
Also note that it only does this calculation when choosing a target. Once it has a target, it only cares about whether the robot is on our side of the field